assets/quick-order-list.js

```diff
@@ -477,18 +456,51 @@ if (!customElements.get('quick-order-list')) {
         }, 1000);
       }
 
-      toggleLoading(id, enable) {
-        const quickOrderListItems = this.querySelectorAll(`#Variant-${id} .loading__spinner`);
-        const quickOrderListItem = this.querySelector(`#Variant-${id}`);
+      toggleLoading(loading, target = this) {
+        target.querySelector('#shopping-cart-variant-item-status').toggleAttribute('aria-hidden', !loading);
+        target.querySelector('.variant-remove-total .loading__spinner')?.classList.toggle('hidden', !loading);
+      }
+    }
+  );
+}
 
-        if (enable) {
-          quickOrderListItem.classList.add('quick-order-list__container--disabled');
-          [...quickOrderListItems].forEach((overlay) => overlay.classList.remove('hidden'));
-          this.variantItemStatusElement.setAttribute('aria-hidden', false);
-        } else {
-          quickOrderListItem.classList.remove('quick-order-list__container--disabled');
-          quickOrderListItems.forEach((overlay) => overlay.classList.add('hidden'));
-        }
+if (!customElements.get('quick-order-list-remove-all-button')) {
+  customElements.define(
+    'quick-order-list-remove-all-button',
+    class QuickOrderListRemoveAllButton extends HTMLElement {
+      constructor() {
+        super();
+        this.quickOrderList = this.closest('quick-order-list');
+
+        this.actions = {
+          confirm: 'confirm',
+          remove: 'remove',
+          cancel: 'cancel',
+        };
+
+        this.addEventListener('click', (event) => {
+          event.preventDefault();
+          if (this.dataset.action === this.actions.confirm) {
+            this.toggleConfirmation(false, true);
+          } else if (this.dataset.action === this.actions.remove) {
+            const items = this.quickOrderList.cartVariantsForProduct.reduce(
```

Comment:
This class was moved down from the start of the file. Only other real change is that we grab the total cart contents from `this.quickOrderList.cartVariantsForProduct` rather than reading from the DOM.